packages/bridge-ui-v2/src/libs/connect/web3modal.ts

```diff
@@ -1,57 +1,51 @@
 
-export const web3modal = new Web3Modal(
-  {
-    projectId,
-    chainImages,
-    // TODO: can we bring these vars into Tailwind theme?
-    themeVariables: {
-      '--w3m-font-family': '"Public Sans", sans-serif',
-      '--w3m-font-feature-settings': 'normal',
-      '--w3m-button-border-radius': '9999px',
-
-      // Body small regular
-      '--w3m-text-small-regular-size': '14px',
-      '--w3m-text-small-regular-weight': '400',
-      '--w3m-text-small-regular-line-height': '20px',
-
-      // Body regular
-      '--w3m-text-medium-regular-size': '16px',
-      '--w3m-text-medium-regular-weight': '400',
-      '--w3m-text-medium-regular-line-height': '24px',
-      '--w3m-text-medium-regular-letter-spacing': 'normal',
-
-      // Title body bold
-      '--w3m-text-big-bold-size': '18px',
-      '--w3m-text-big-bold-weight': '700',
-      '--w3m-text-big-bold-line-height': '24px',
-
-      '--w3m-background-color': 'var(--primary-brand)',
-      '--w3m-overlay-background-color': 'var(--overlay-background)',
-      '--w3m-background-border-radius': '20px',
-      '--w3m-container-border-radius': '0',
-
-      // Unofficial variables
-      // @ts-ignore
-      '--w3m-color-fg-1': 'var(--primary-content)',
-      '--w3m-color-bg-1': 'var(--primary-background)',
-      '--w3m-color-bg-2': 'var(--neutral-background)',
-      '--w3m-color-overlay': 'none',
-      '--w3m-accent-color': 'var(--neutral-background)',
-      '--w3m-accent-fill-color': 'var(--dark-background)',
-    },
-    themeMode: (localStorage.getItem('theme') as 'dark' | 'light') ?? 'dark',
+export const web3modal = createWeb3Modal({
+  wagmiConfig,
+  projectId,
+  chains,
+  chainImages,
+  themeVariables: {
+    '--w3m-font-family': '"Public Sans", sans-serif',
+
+    // Body small regular
+    // @ts-ignore
+    '--w3m-text-small-regular-line-height': '20px',
+
+    // Body regular
+    // @ts-ignore
+    '--w3m-text-medium-regular-size': '16px',
+    '--w3m-text-medium-regular-weight': '400',
+    '--w3m-text-medium-regular-line-height': '24px',
+    '--w3m-text-medium-regular-letter-spacing': 'normal',
+
+    // Title body bold
+    // @ts-ignore
+    '--w3m-text-big-bold-size': '18px',
+    '--w3m-text-big-bold-weight': '700',
+    '--w3m-text-big-bold-line-height': '24px',
+
+    // @ts-ignore
+    '--w3m-background-color': 'var(--primary-brand)',
+    '--w3m-overlay-background-color': 'var(--overlay-background)',
+    '--w3m-background-border-radius': '20px',
+    '--w3m-container-border-radius': '0',
+
+    // Unofficial variables
+    // @ts-ignore
+    '--w3m-color-fg-1': 'var(--primary-content)',
+    '--w3m-color-bg-1': 'var(--primary-background)',
+    '--w3m-color-bg-2': 'var(--neutral-background)',
+    '--w3m-color-overlay': 'none',
+    '--w3m-accent-color': 'var(--neutral-background)',
+    '--w3m-accent-fill-color': 'var(--dark-background)',
   },
-  ethereumClient,
-);
+  themeMode: (localStorage.getItem('theme') as 'dark' | 'light') ?? 'dark',
```

Comment:
The theme switching does not work correctly. It stays on the initial one until you refresh the page.

<img width="522" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/5267230/6f895290-2884-4657-8855-79cbc26763d1">

Reply:
Ok, so it is possible as per the demo:
<img width="447" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/5267230/92580d56-dd23-4057-ae5e-6d9b4c2c0447">

It would also be great to combine the two "buttons" we have now manually and use the combined one from walletconnect again. Assuming they fixed the hook to update the balance.

Reply:
Good catch. Updated https://github.com/taikoxyz/taiko-mono/pull/14999/files#diff-555d989884c6cc2a88ae4fde18e1e462fc54d9c57c6666ee244fcee1cc21d837L13 to resolve it.